packages/synapse-react-client/src/components/AccessRequirementList/ManagedACTAccessRequirementRequestFlow/DataAccessRequestAccessorsFilesForm.tsx

```diff
@@ -145,21 +157,128 @@ export default function DataAccessRequestAccessorsFilesForm(
     })
   }
 
-  const { mutate: submit } = useSubmitDataAccessRequest({
-    onSuccess: () => {
-      onSubmissionCreated()
-    },
-    onError: onError,
-  })
+  const { mutate: submit, isLoading: isLoadingSubmitDataAccessRequest } =
+    useSubmitDataAccessRequest({
+      onSuccess: () => {
+        onSubmissionCreated()
+      },
+      onError: onError,
+    })
 
-  const { mutateAsync: updateRequestAsync } = useUpdateDataAccessRequest({
+  const {
+    mutateAsync: updateRequestAsync,
+    isLoading: isLoadingUpdateDataAccessRequest,
+  } = useUpdateDataAccessRequest({
     onError: onError,
   })
 
+  const isLoading =
+    isLoadingGetDataAccessRequest ||
+    isLoadingUpdateDataAccessRequest ||
+    isLoadingSubmitDataAccessRequest
+
+  /**
+   * This effect comprises a collection of updates we should immediately apply to a data access request.
+   */
+  useEffect(() => {
+    if (dataAccessRequest && user) {
+      let shouldUpdate = false
+
+      // Attach the researchProjectId to the request
+      if (!dataAccessRequest.researchProjectId) {
+        dataAccessRequest.researchProjectId = researchProjectId
+        shouldUpdate = true
+      }
+
+      // Add the current user with GAIN_ACCESS to the list of accessors
+      const currentUserWithGainAccess: AccessorChange = {
+        userId: user.ownerId,
+        type: isRenewal ? AccessType.RENEW_ACCESS : AccessType.GAIN_ACCESS,
+      }
+      if (
+        !dataAccessRequest.accessorChanges ||
+        !dataAccessRequest.accessorChanges.find(item =>
+          deepEquals(item, currentUserWithGainAccess),
+        )
+      ) {
+        dataAccessRequest.accessorChanges = [
+          currentUserWithGainAccess,
+          ...(dataAccessRequest.accessorChanges || []),
+        ]
+        shouldUpdate = true
+      }
+
+      // SWC-5765: Filter out duplicate accessors
+      const seen = new Set()
+      const uniqueAccessorChanges = dataAccessRequest.accessorChanges.filter(
+        accessorChange => {
+          return seen.has(accessorChange.userId)
+            ? false
+            : seen.add(accessorChange.userId)
+        },
+      )
+      if (
+        uniqueAccessorChanges.length !==
+        dataAccessRequest.accessorChanges.length
+      ) {
+        dataAccessRequest.accessorChanges = uniqueAccessorChanges
+        shouldUpdate = true
+      }
+
+      if (shouldUpdate) {
+        updateRequestAsync(dataAccessRequest)
```

Comment:
May need to clear/reset vs invalidate query to ensure we enter a loading state until we get the new etag. Might be fine, if we set the query data based on the response from the update.